### DISCUSSION TITLE
Unrecognized configuration parameter "jwt.claims.user_id - Database function

### DISCUSSION THREAD

**Discussion by ethanfox** [supabase/supabase#16627] [Questions]
I am trying to create a database function that helps convert a contact to a user. My app is for designers and they create contacts within the app and then share documents with contacts. An email is sent to the contact's email and then they register and now become a user. This database function serves to aid that transition by linking contacts to a particular user. I am using Sveltekit

However, I receive this error:
code:"42704"
details: null
hint: null
message: "unrecognized configuration parameter \"jwt.claims.user_id\""

Im not sure what the jwt serves in this. I'm not calling for it so I assume it is coming from the client. I have disabled RLS on my profiles table and I still get the same error.
My user table is called profiles.

```
DATABASE FUNCTION

declare
    v_user_id uuid; -- Renamed variable
begin
    select
        id into v_user_id -- Use the renamed variable here
    from
        profiles -- Changed table name to "profiles"
    where
        email = contact_email;

    if v_user_id is not null then
        update
            contacts
        set
            user_id = v_user_id -- And here
        where
            email = contact_email;

        update
            shared_documents sd
        set
            recipient_id = v_user_id -- And here
        from
            contacts c
        where
            sd.recipient_id = c.id
            and c.email = contact_email
            and sd.is_contact = true;
    end if;
end;
```

```
FRONTEND FUNCTION:
async function convertContact(contactEmail) {
		console.log("Converting contact", contactEmail);
		try {
			const { data, error } = await pageData.supabase.rpc("convert_contact", {
				contact_email: contactEmail
			});

			if (error) throw error;

			// Update the state with the response data
			console.log("STATUS:", data);
		} catch (error) {
			console.error("Failed to Convert Contact", error);
		}
	}
```

**Comment by GaryAustin1** [comment]
Do you have user_id in any RLS on the tables being accessed (not just profiles).  I see contacts at least.  Do you use another auth system that creates a claim user_id in the jwt you pass to Supabase client.  This is not a standard Supabase claim.

Just ideas, not really sure where it is coming from.

**Reply by ethanfox** [reply]
No I only use the term "user_id" in the contact table in order to refer to the connected user. For the rest of the tables I simply use id for the uuid column. I have no other auth system - only Supabase. Appreciate the help.

I feel like it might be coming from `pageData.supabase` but I'm not sure if a JWT would be passed if calling a database function from the frontend.

**Reply by GaryAustin1** [reply]
A jwt is passed on every REST API call to the database even rpc calls.  I would think something though is doing auth.jwt()->>user_id or using jwt.claims.user_id in your code/RLS somewhere in order to get that error.  The API does not check if you have extra claims that I know of.
